pkg/kudoctl/packages/package_manager.go

```diff
@@ -44,69 +45,62 @@ func ReadPackage(fs afero.Fs, path string) (Package, error) {
 	if err != nil {
 		return nil, err
 	}
+	clog.V(4).Printf("determining package type of %v", path)
 	// order of discovery
 	// 1. tarball
 	// 2. file based
 	if fi.Mode().IsRegular() && strings.HasSuffix(path, ".tgz") {
-		r, err := getFileReader(fs, path)
+		b, err := afero.ReadFile(fs, path)
 		if err != nil {
 			return nil, err
 		}
-
-		return tarPackage{r}, nil
+		buf := bytes.NewBuffer(b)
+		clog.V(4).Printf("%v is a tar package", path)
+		return tarPackage{buf}, nil
 	} else if fi.IsDir() {
+		clog.V(4).Printf("%v is a file package", path)
 		return filePackage{path, fs}, nil
 	} else {
 		return nil, fmt.Errorf("unsupported file system format %v. Expect either a *.tgz file or a folder", path)
 	}
 }
 
-func getFileReader(fs afero.Fs, path string) (io.Reader, error) {
-	f, err := fs.Open(path)
-	if err != nil {
-		return nil, err
-	}
-	return f, nil
-}
-
-// NewPackageFromReader is a package from a reader.  This should only be used when a file cache isn't used.
-func NewPackageFromReader(r io.Reader) Package {
-	return tarPackage{r}
+// NewPackageFromBytes is a package from a reader.  This should only be used when a file cache isn't used.
```

Comment:
we're not using reader anymore

Reply:
Do we even need this method? I always doubt the reasoning behind one line methods... if that is not expressive enough that we have the feel to wrap it in a named method then maybe we have chosen wrong names for that object itself?